tests/unit/test_charm.py

```diff
@@ -161,6 +184,36 @@ def test_layer_updated_without_any_endpoint_info(harness: Harness) -> None:
     assert harness.charm._login_ui_layer.to_dict() == expected_layer
 
 
+def test_layer_updated_with_tracing_endpoint_info(harness: Harness) -> None:
+    """Test Pebble Layer when relation data is in place."""
+    harness.set_leader(True)
+    harness.set_can_connect(CONTAINER_NAME, True)
+    harness.charm.on.login_ui_pebble_ready.emit(CONTAINER_NAME)
+    tracing_relation_id = setup_tracing_relation(harness)
+
+    host = harness.get_relation_data(tracing_relation_id, "tempo-k8s")["host"]
+    http = filter(
+        lambda x: x["protocol"] == "otlp_http",
+        harness.get_relation_data(tracing_relation_id, "tempo-k8s")["ingesters"],
+    )
+    grpc = filter(
+        lambda x: x["protocol"] == "otlp_grpc",
+        harness.get_relation_data(tracing_relation_id, "tempo-k8s")["ingesters"],
+    )
+    assert (
+        harness.charm._login_ui_layer.to_dict()["services"][CONTAINER_NAME]["environment"][
+            "OTEL_HTTP_ENDPOINT"
+        ]
+        == f"{host}:{http}"
+    )
+    assert (
+        harness.charm._login_ui_layer.to_dict()["services"][CONTAINER_NAME]["environment"][
```

Comment:
The same could be avoided here. We have multiple calls to the `harness.charm._login_ui_layer.to_dict()["services"][CONTAINER_NAME]["environment"]`.